statsforecast/core.py

```diff
@@ -1,121 +1,233 @@
-# AUTOGENERATED! DO NOT EDIT! File to edit: nbs/core.ipynb (unless otherwise specified).
+# AUTOGENERATED! DO NOT EDIT! File to edit: ../nbs/core.ipynb.
 
+# %% auto 0
 __all__ = ['StatsForecast']
 
-# Cell
+# %% ../nbs/core.ipynb 4
 import inspect
 import logging
+from copy import deepcopy
 from functools import partial
 from os import cpu_count
 from typing import Any, Callable, List, Optional, Tuple
 
 import numpy as np
 import pandas as pd
 
-# Internal Cell
+# %% ../nbs/core.ipynb 5
 logging.basicConfig(
     format='%(asctime)s %(name)s %(levelname)s: %(message)s',
     datefmt='%Y-%m-%d %H:%M:%S',
 )
 logger = logging.getLogger(__name__)
 
-# Internal Cell
+# %% ../nbs/core.ipynb 8
 class GroupedArray:
-
+    
     def __init__(self, data, indptr):
         self.data = data
         self.indptr = indptr
         self.n_groups = self.indptr.size - 1
-
+        
     def __getitem__(self, idx):
         if isinstance(idx, int):
             return self.data[self.indptr[idx] : self.indptr[idx + 1]]
         elif isinstance(idx, slice):
             idx = slice(idx.start, idx.stop + 1, idx.step)
             new_indptr = self.indptr[idx].copy()
-            new_data = self.data[new_indptr[0] : new_indptr[-1]].copy()
+            new_data = self.data[new_indptr[0] : new_indptr[-1]].copy()            
             new_indptr -= new_indptr[0]
             return GroupedArray(new_data, new_indptr)
         raise ValueError(f'idx must be either int or slice, got {type(idx)}')
-
+    
     def __len__(self):
         return self.n_groups
-
+    
     def __repr__(self):
         return f'GroupedArray(n_data={self.data.size:,}, n_groups={self.n_groups:,})'
-
+    
     def __eq__(self, other):
         if not hasattr(other, 'data') or not hasattr(other, 'indptr'):
             return False
         return np.allclose(self.data, other.data) and np.array_equal(self.indptr, other.indptr)
-
-    def compute_forecasts(self, h, func, xreg=None, fitted=False, level=None, *args):
-        has_level = 'level' in inspect.signature(func).parameters and level is not None
-        if has_level:
-            out = np.full((h * self.n_groups, 2 * len(level) + 1), np.nan, dtype=np.float32)
-            func = partial(func, level=level)
-        else:
-            out = np.full(h * self.n_groups, np.nan, dtype=np.float32)
-        if fitted:
-            fitted_vals = np.full(self.data.shape[0], np.nan, dtype=np.float32)
-        xr = None
-        keys = None
+    
+    def fit(self, models):
```

Comment:
I suggest to keep the StatsForecast class methods to depend on inputs with the `_df` suffix.
Example: X numpy array vs X_df pandas dataframe

This will make users easier to distinguish between the lower-level APIs that depend on numpy and higher level APIs that depend on dataframes.